course3/UniquePtr/my_unique_ptr.h

```diff
@@ -0,0 +1,221 @@
+#include <iostream>
+/*
+re-implemention of http://en.cppreference.com/w/cpp/memory/unique_ptr
+*/
+
+template<class T>
+class MyUniquePtr
+{
+public:
+    MyUniquePtr() = default;
+    
+    ~MyUniquePtr(){}
+    
+    MyUniquePtr(T*&& ptr) : mData(ptr)
+    {
+        delete ptr;
+    }
+    
+    MyUniquePtr(T& ptr) : mData(ptr)
+    {
+        mData = ptr.mData;
+    }
+    
+    // move
+    MyUniquePtr(T&& ptr)
+    {
+        std::move(mData, ptr.mData);
+    }
+    
+    
+    MyUniquePtr& operator=(T& other)
+    {
+        mData = other.mData;
+    }
+
+
+    T *get() const
```

Comment:
Please, stick the ```*``` to the type name, as it is a part of the type, not the method name:
```
T* get() const
```